components/apitemplate_io/app/apitemplate_io.app.ts

```diff
@@ -1,13 +1,279 @@
+      ],
+    },
+    templateId: {
+      type: "string",
+      label: "Template Id",
+      description: "The ID of the template that you want to use.",
+      async options({
+        prevContext,
+        format,
+      }) {
+        const limit = 300;
+        let offset = 0;
+        if (prevContext && Number.isInteger(prevContext.offset)) {
+          offset = ((prevContext.offset / limit) + 1) * limit;
+        }
+        return this.getTemplateOpts(limit, offset, format);
+      },
+    },
+    transactionRef: {
+      type: "string",
+      label: "Transaction Ref",
+      description: "Object transaction reference.",
+      async options({ prevContext }) {
+        const limit = 300;
+        let offset = 0;
+        if (prevContext && Number.isInteger(prevContext.offset)) {
+          offset = ((prevContext.offset / limit) + 1) * limit;
+        }
+        return this.listObjectOpts(limit, offset);
+      },
+    },
+    expiration: {
+      type: "integer",
+      label: "Expiration",
+      description: "Expiration of the generated image in minutes(default to `0`, store permanently).",
+      optional: true,
+    },
+    meta: {
+      type: "string",
+      label: "Meta",
+      description: "Specify an external reference ID for your own reference. It appears in the `list-objects` API.",
+      optional: true,
+    },
+    overrides: {
+      type: "any",
+      label: "Overrides",
+      description: "Array of objects with the property name and the replaced values on the template, as shown below:\n\n`[{\"name\": \"text_name\",\"text\": \"My Name\"}, {\"name\": \"text_quote\",\"text\": \"Lorem ipsum dolor sit...\"}]`",
+    },
+    data: {
+      type: "any",
+      label: "Data",
+      description: "Object to replace values on the template, you can [check here](https://app.apitemplate.io/manage-templates/) for the JSON example for your template.",
+    },
+  },
   methods: {
-    // this.$auth contains connected account data
-    authKeys() {
-      console.log(Object.keys(this.$auth));
+    _getUrl(api, path) {
+      if (api) {
+        return `${api}/v2${path}`;
+      }
+      return `https://rest.apitemplate.io/v2${path}`;
+    },
+    _getHeaders(headers = {}) {
+      return {
+        "Authorization": this.$auth.api_key,
+        "Content-Type": "application/json",
+        ...headers,
+      };
+    },
+    _getRequestParams(opts: any) {
+      return {
+        ...opts,
+        url: this._getUrl(opts.api, opts.path),
+        headers: this._getHeaders(),
+      };
+    },
+    async getTemplates($ = this, limit, offset, format) {
+      const response = await axios($, this._getRequestParams({
+        method: "GET",
+        path: "/list-templates",
+        params: {
+          limit,
+          offset,
+          format,
+        },
+      }));
+      return response;
+    },
+    async getTemplateOpts(limit, offset, format) {
+      const { templates } = await this.getTemplates(this, limit, offset, format);
```

Comment:
```suggestion
      const { templates } = await this.getTemplates(this, params: { limit, offset, format });
```
How about passing parameters in `params` to avoid repetition?